src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/ApiController.cs

```diff
@@ -114,9 +114,12 @@ public async Task<IActionResult> Create(CreateApiViewModel model)
             shellSettings["RecipeName"] = model.RecipeName;
             shellSettings["FeatureProfile"] = model.FeatureProfile;
 
-            model.IsNewTenant = true;
+            if (ModelState.IsValid)
+            {
+                model.IsNewTenant = true;
```

Comment:
Is there a reason why this needs to be set to true, and we can't just use the model's `IsNewTenant`, which came from the request? In the end, this is the same as before regarding what's passed to the `_tenantvalidator`, and still, I don't think [this code](https://github.com/OrchardCMS/OrchardCore/blob/07698e42aa295b6be7aebf38e18b4a9a2c31d6a5/src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantValidator.cs#L110) is ever reached if the model's `IsNewTenant` is set to true. 
So in my opinion, this line of code should not exist.

The use case would be to be able to send a request to the `Create()` endpoint, and after that, anytime we could send the same model, just with `IsNewTenant = false`, and get a new setup token since this could expire.

Regarding the test, I think It should contain 2 requests, they should be identical, except that in the second request's model, `IsNewTenant` should be set to false.

Just to give a bit more context, for testing purposes I have used our [ApiCLient](https://github.com/Lombiq/Orchard-Core-API-Client).

Reply:
> Is there a reason why this needs to be set to true, and we can't just use the model's IsNewTenant, which came from the request?

Don't TRUST such flag from the client-side, the intend of `IsNewTenant` If I remember is to distinguish between the create and edit tenant

> The use case would be to be able to send a request to the Create() endpoint, and after that, anytime we could send the same model, just with IsNewTenant = false, and get a new setup token since this could expire.

We could write a unit test for this too

> Regarding the test, I think It should contain 2 requests, they should be identical, except that in the second request's model, IsNewTenant should be set to false.

> Just to give a bit more context, for testing purposes I have used our [ApiCLient](https://github.com/Lombiq/Orchard-Core-API-Client).

Thanks for point to your API test client, I will have a look if the time permits

Reply:
FYI

https://github.com/OrchardCMS/OrchardCore/blob/07698e42aa295b6be7aebf38e18b4a9a2c31d6a5/src/OrchardCore.Modules/OrchardCore.Tenants/ViewModels/TenantViewModel.cs#L30-L31

Reply:
I see, thank you.
But then how could this be possible?

> The use case would be to be able to send a request to the Create() endpoint, and after that, anytime we could send the same model, just with IsNewTenant = false, and get a new setup token since this could expire.

If I have created a tenant with an API call, and initiate new calls to the `Create()` endpoint, the validator will always add a model error for the name since this is already existing, but obviously, we don't want to create a new tenant with the same name, just get the existing tenant's setup token.

Reply:
How the API should work is that if you send identical requests, the second should succeed too but just refer to the already existing tenant (with a new token), without having to set flags on the model.

Reply:
This could be done internally in the `ApiController` but again I don't think there's a value to set `IsNewTenant` flag to `false`, coz the `TenantValidatorService` is using this flag to distinguish between create and edit